lsp-completion.el

```diff
@@ -671,53 +681,54 @@ The CLEANUP-FN will be called to cleanup."
   :group 'lsp-mode
   :global nil
   :lighter ""
-  (cond
-   (lsp-completion-mode
-    (setq-local completion-at-point-functions nil)
-    (add-hook 'completion-at-point-functions #'lsp-completion-at-point nil t)
-    (setq-local completion-category-defaults
-                (add-to-list 'completion-category-defaults '(lsp-capf (styles basic))))
-
+  (let ((completion-started-fn (lambda (&rest _)
+                                 (setq-local lsp-inhibit-lsp-hooks t)))
+        (after-completion-fn (lambda (&rest _)
+                               (setq lsp-completion--last-result nil)
+                               (lsp-completion--clear-cache)
+                               (setq-local lsp-inhibit-lsp-hooks nil))))
     (cond
-     ((equal lsp-completion-provider :none))
-     ((and (member lsp-completion-provider '(:capf nil t))
-           (fboundp 'company-mode))
-      (company-mode 1)
-      (when (or (null lsp-completion-provider)
-                (member 'company-lsp company-backends))
-        (lsp--warn "`company-lsp` is not supported anymore. Using `company-capf` as the `lsp-completion-provider`."))
-      (add-to-list 'company-backends 'company-capf))
+     (lsp-completion-mode
+      (setq-local completion-at-point-functions nil)
+      (add-hook 'completion-at-point-functions #'lsp-completion-at-point nil t)
+      (setq-local completion-category-defaults
+                  (add-to-list 'completion-category-defaults '(lsp-capf (styles basic))))
+
+      (cond
+       ((equal lsp-completion-provider :none))
+       ((and (member lsp-completion-provider '(:capf nil t))
+             (fboundp 'company-mode))
+        (company-mode 1)
+        (when (or (null lsp-completion-provider)
+                  (member 'company-lsp company-backends))
+          (lsp--warn
+           "`company-lsp' is not supported anymore.  Using `company-capf' as the `lsp-completion-provider'."))
+        (add-to-list 'company-backends 'company-capf))
+       (t
+        (lsp--warn "Unable to autoconfigure company-mode.")))
+
+      (when (bound-and-true-p company-mode)
+        (add-hook 'company-completion-started-hook
+                  completion-started-fn
+                  nil
+                  t)
+        (add-hook 'company-after-completion-hook
```

Comment:
We had issues with these hooks in the past, right? Dont we have to us the other mechanism that crosschecks if the completion cache is relevant to the current position?

Reply:
Yes, I've updated that part too. Thanks